libcontainer/nsenter/cloned_binary.c

```diff
@@ -415,32 +513,51 @@ static int try_bindfd(void)
 	close(fd);
 
 	/*
-	 * For obvious reasons this won't work in rootless mode because we haven't
-	 * created a userns+mntns -- but getting that to work will be a bit
-	 * complicated and it's only worth doing if someone actually needs it.
+	 * Daemons such as systemd and udisks2 watch /proc/self/mountinfo and
+	 * re-parse it on every change, which gets expensive when the mount table
+	 * is large and/or changes frequently. Set up the bind-mounts in a new,
```

Comment:
> the bind-mounts

Nit: we only have one single bind mount.

Reply:
Right; that's poorly worded. There is only one bind mount, but three mutations to the mount table. The kernel notifies those daemons for each `mount` and `umount2` syscall occuring in the mount namespace.

Reply:
Perhaps something like `s/Set up the bind-mounts in/Perform the mounts in/`